ompi/runtime/ompi_rte.c

```diff
@@ -818,6 +818,11 @@ int ompi_rte_init(int *pargc, char ***pargv)
         opal_process_info.initial_wdir = val;
         val = NULL;  // protect the string
     }
+    else {
+        // Probably singleton case. Just assume cwd.
+        opal_process_info.initial_wdir = calloc(1, OPAL_PATH_MAX + 1);
+        getcwd(opal_process_info.initial_wdir, OPAL_PATH_MAX);
```

Comment:
you may want to use opal_getcwd here instead

Reply:
thanks - fixed